src/workers/newView.ts

```diff
@@ -159,13 +142,13 @@ const worker: Worker = {
       throw err;
     }
 
-    // no need to touch reading streaks if we didn't save a new view event or if we don't have a userId
-    if (!didSave || !data.userId) {
+    // no need to touch reading streaks if we didn't save a new view or if we don't have a userId
+    if (!didSave || !data.userId || !latestView) {
```

Comment:
Why do we need the `!latestView` condition as well? If `didSave` is `true`, `latestView` will always be defined.

If you want, we can instead base the logic on the `!latestView`. But then we can get rid of the `didSave` variable completely and rely on `latestView` being non null instead.